src/StackExchange.Redis/ResultProcessor.cs

```diff
@@ -1262,6 +1265,22 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
             }
         }
 
+        private sealed class BooleanArrayProcessor : ResultProcessor<Boolean[]>
+        {
+            protected override bool SetResultCore(PhysicalConnection connection, Message message, in RawResult result)
+            {
+                switch (result.Type)
+                {
+                    case ResultType.MultiBulk:
+                        var arr = result.GetItemsAsBooleans()!;
```

Comment:
This assertion is blind, so a null result would go boom - be sure to add a check, e.g.:
```suggestion
                    case ResultType.MultiBulk when !result.IsNull:
                        var arr = result.GetItemsAsBooleans()!;
```

But I'm not sure this is correct. We should consider the case the key is missing, and add tests for that in a few of these PRs to ensure we get the expected result.